### PR TITLE
Add some methods to the URLUtil

### PR DESCRIPTION
**Pull request by dnlkoch**
This adds the following methods to the `URLUtil`:
  * getQueryParam()
  * hasQueryParam()
  * createValidGetCapabilitiesRequest()
  * isValid()

### CODE REVIEW

File: src/Util/UrlUtil/UrlUtil.spec.js [hunk 1]
```diff
@@ -73,6 +73,22 @@ describe('UrlUtil', () => {
...
+      it('returns the value of the given query param of the provided URL', () => {
+        let got = UrlUtil.getQueryParam('http://borussia.de/bvb', 'bvb');
+        expect(got).toEqual(undefined);
+
+        got = UrlUtil.getQueryParam('http://borussia.de/bvb?bvb=09', 'bvb');
+        expect(got).toEqual('09');
+
+        got = UrlUtil.getQueryParam('http://borussia.de/bvb?bvb=09&shinji=kagawa', 'shinji');
+        expect(got).toEqual('kagawa');
+      });
```
Comment (marcjansen) [inline src/Util/UrlUtil/UrlUtil.spec.js]: What do we get back from

```
UrlUtil.getQueryParam('http://example.com/foo?humpty=1&humpty=2', 'humpty')
```
?

File: src/Util/UrlUtil/UrlUtil.spec.js [hunk 2]
```diff
@@ -151,6 +211,28 @@ describe('UrlUtil', () => {
...
+      it('validates the given URL', () => {
+        const urls = {
+          'http://ows.terrestris.de': true,
+          'https://ows.terrestris.de': true,
+          'ftp://ows.terrestris.de': true,
+          'http://ows.terrestris.de:8080': true,
+          'http://ows.terrestris.de?param=key': true,
+          'http://ows.terrestris.de?param1=key1&param2=key2': true,
+          'http://10.133.7.9': true,
```
Comment (marcjansen) [inline src/Util/UrlUtil/UrlUtil.spec.js]: how about testing

```
'http://foo@example.com/foo?humpty=dumpty',
'http://foo:s3cretP4ssw0rd@example.com/foo?humpty=dumpty'
```

do we allow these? I guess the rest of special cases like umlauts and encoded params are handled in the validator library, right?

File: src/Util/UrlUtil/UrlUtil.spec.js (hunk 1 above)
Comment (dnlkoch) [reply src/Util/UrlUtil/UrlUtil.spec.js]: Actually it returns `2` as the query string gets parsed by the underlying modules `url-parse`/`querystringify` to `{ humpty: '2' }`.

File: src/Util/UrlUtil/UrlUtil.js [hunk 3]
```diff
@@ -84,6 +100,46 @@ export class UrlUtil {
...
+    const queryParamsObj = UrlUtil.getQueryParams(url);
+
+    return !!Object.keys(queryParamsObj).some(k => k.toLowerCase() === key.toLowerCase());
+  }
+
+  /**
+   * Creates a valid GetCapabilitiesRequest out of the given URL by checking if
+   * SERVICE, REQUEST and VERSION are set.
+   *
+   * @param {String} url The URL to validate. Default is to 'WMS'.
```
[the remark this answers is not in the document]
Comment (dnlkoch) [reply src/Util/UrlUtil/UrlUtil.js]: Correct, fixed in 2aaa201.

File: src/Util/UrlUtil/UrlUtil.spec.js (hunk 2 above)
Comment (dnlkoch) [reply src/Util/UrlUtil/UrlUtil.spec.js]: Yes, both urls are allowed, test cases added in 5dc63fe.

File: src/Util/UrlUtil/UrlUtil.spec.js (hunk 1 above)
Comment (marcjansen) [reply src/Util/UrlUtil/UrlUtil.spec.js]: OK. I *think* a return value of `[1, 2]` would be more correct. See e.g. this library: https://github.com/sindresorhus/query-string.

**Comment by marcjansen** [comment]
One other question, shouldn't we use `decodeURIComponent` on the values before returning them? Or is this handled elsewhere?

File: src/Util/UrlUtil/UrlUtil.spec.js (hunk 2 above)
Comment (marcjansen) [reply src/Util/UrlUtil/UrlUtil.spec.js]: :+1:

File: src/Util/UrlUtil/UrlUtil.spec.js (hunk 1 above)
Comment (dnlkoch) [reply src/Util/UrlUtil/UrlUtil.spec.js]: Yes, I agree. Switched from `querystringify` to `query-string` to get this done. See ca71de7.

**Comment by dnlkoch** [comment]
Thx for review @marcjansen. Will merge this when travis is happy.

**Pull request merged**